atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/workflow/WriteOnceDeleteOnceWorkflowTest.java

```diff
@@ -0,0 +1,145 @@
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.IsolationLevel;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedDeleteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedWriteTransactionAction;
+import com.palantir.atlasdb.workload.util.AtlasDbUtils;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.IntFunction;
+import org.junit.Test;
+
+public class WriteOnceDeleteOnceWorkflowTest {
+    private static final String TABLE_NAME = "my.latte";
+    private static final WriteOnceDeleteOnceWorkflowConfiguration CONFIGURATION =
+            ImmutableWriteOnceDeleteOnceWorkflowConfiguration.builder()
+                    .tableConfiguration(ImmutableTableConfiguration.builder()
+                            .tableName(TABLE_NAME)
+                            .isolationLevel(IsolationLevel.SERIALIZABLE)
+                            .build())
+                    .rateLimit(Double.MAX_VALUE)
+                    .iterationCount(1)
+                    .build();
+
+    private final InteractiveTransactionStore memoryStore = AtlasDbTransactionStore.create(
+            TransactionManagers.createInMemory(ImmutableSet.of()),
+            ImmutableMap.of(
+                    TableReference.createWithEmptyNamespace(TABLE_NAME),
+                    AtlasDbUtils.tableMetadata(IsolationLevel.SERIALIZABLE)));
+
+    @Test
+    public void deleteWhenBuggifyTrue() {
+        Workflow workflow = createWorkflow(AlwaysBuggifyFactory.INSTANCE, i -> i);
+        WorkflowHistory history = workflow.run();
+        WitnessedTransaction transaction = Iterables.getOnlyElement(history.history());
+        WitnessedTransactionAction action = Iterables.getOnlyElement(transaction.actions());
+        assertThat(action).isInstanceOf(WitnessedDeleteTransactionAction.class);
+    }
+
+    @Test
+    public void writeWhenBuggifyFalse() {
+        Workflow workflow = createWorkflow(NoOpBuggifyFactory.INSTANCE, i -> i);
+        WorkflowHistory history = workflow.run();
+        WitnessedTransaction transaction = Iterables.getOnlyElement(history.history());
+        WitnessedTransactionAction action = Iterables.getOnlyElement(transaction.actions());
+        assertThat(action).isInstanceOf(WitnessedWriteTransactionAction.class);
+    }
+
+    @Test
+    public void calculateChanceForDeleteHasExpectedValues() {
+        assertThat(WriteOnceDeleteOnceWorkflows.calculateChanceForDelete(0, 100))
+                .isEqualTo(0.01);
+        assertThat(WriteOnceDeleteOnceWorkflows.calculateChanceForDelete(99, 100))
+                .isEqualTo(1.00);
+    }
+
+    @Test
+    public void writesThenDeletes() {
+        Iterator<Integer> keys = List.of(0, 0, 1, 1).iterator();
```

Comment:
Probably should just do `Iterators.cycle`, but at the same time this is a bit easier to understand.

Reply:
The other sequence is valid too right? (write 0, delete 1, delete 0, write 1)

Reply:
Yup of course, I was just modeling this to be closer to how it would be in practice, which would not be write-delete-write-delete